src/tasks/beacon/beacon_task.c

```diff
@@ -118,12 +61,9 @@ void beacon_task_dispatch(slate_t *slate)
     pkt.flags = 0;
     pkt.seq = 0;
 
-    // Serialize data from slate variables
-    // Current state name
-    size_t pkt_len = serialize_slate(slate, tmp_data);
-
     // Commit into serialized byte array
-    pkt.len = pkt_len;
+    memset(tmp_data, 0, sizeof(tmp_data));
+    pkt.len = serialize_slate(slate, tmp_data);
```

Comment:
Hmm why we choose to not write directly into `pkt.data` here?

Reply:
Yes, I think we can directly do that.